### ISSUE TITLE
headings (and possibly other formatting?) don't work within snap

### ISSUE THREAD

**Issue by joachim-n** [gitpitch/gitpitch#197]
This doesn't get formatted as a heading:

```
@snap[north]

# One

@snapend
```

See demo here: https://gitpitch.com/joachim-n/git-pitch-demo/master

**Comment by gitpitch** [comment]
The [Snap Layouts Content Types](https://gitpitch.com/docs/layout-features/snap-layout-content-types/) section in the documentation explains why the markdown in your sample slide is not formatted. And what you will need to do to get the desired result.

I recommend reading through the complete [Layout Features](https://gitpitch.com/docs/layout-features) chapter in the docs as it provides lots of useful information and examples to teach you how to get the most out of _snap-layouts_.

**Comment by joachim-n** [comment]
Ah, I hadn't read that far in the docs, sorry!

That does seem really weird though to have these snap sections so limited. Having to use HTML tags within them seems to me to go against the whole purpose of using Markdown in the first place.

**Comment by gitpitch** [comment]
> Ah, I hadn't read that far in the docs, sorry!

No worries.

> That does seem really weird though to have these snap sections so limited. Having to use HTML tags > within them seems to me to go against the whole purpose of using Markdown in the first place.

Thanks for this feedback. I'm just going to jot down a few thoughts and clarifications in response, for you and for any future readers of this thread.

This [one limitation](https://gitpitch.com/docs/layout-features/snap-layout-content-types/) - is (1) real and (2) agreed, not ideal. 

But in truth the practical limitations are minimal - only basic markdown text formatting is not supported within the snap tags. And the workaround is to use the corresponding HTML snippets - which are themselves pretty simple and [perfectly valid](https://github.github.com/gfm/#raw-html) within GitHub Flavored Markdown.

Support for [GitPitch markdown shortcuts](https://gitpitch.com/docs/layout-features/examples-markdown-shortcuts/) within the snap tags should go a long way to mitigating concerns about overall usefulness. The [detailed examples](https://gitpitch.com/docs/layout-features/examples-snap-inspirations/) in the docs try to demonstrate that as best they can. But play around with it and you will see this for yourself.

Bottom line, if you embrace the constraints - that's a real thing apparently :) - you can achieve a whole lot with snap layouts. But they are entirely optional, so I understand they might not be for everyone.

Finally, I will update this thread if I do find a way to seamlessly support auto-formatting for markdown text within snap tags. So stay tuned.

**Comment by joachim-n** [comment]
Thanks for the detailed reply!

I'll explain my use case to give you a bit more of an idea of why this seems like a limitation to me.

I wanted to the same H1 heading appear in the same place over several slides. Because the content below the slides is not the same size, the heading moves up and down which ruins the effect of having a constant heading while the contents below change.

Therefore, snap layout seemed a good solution for this, until I found that the Markdown for headings doesn't work there.

**Comment by gitpitch** [comment]
Again, thanks for this feedback.

> I wanted to the same H1 heading appear in the same place over several slides. Because the content below the slides is not the same size, the heading moves up and down which ruins the effect of having a constant heading while the contents below change.

At this point I'm not sure if you concerns are theoretical or if you have actually tried to use snap-layouts and you experiencing the problem you describe here - headings moving up and down? If you have actually experienced this problem can you please share a link to your sample presentation? I would really like to take a closer look at what you are doing.

I ask these questions because what you describe should be impossible. Should be. The only way I can imagine snapped-content moving on a slide (even temporarily) is if you have not taken the required step to deactivate transitions on the slide, per the [snap-layouts documentation](https://gitpitch.com/docs/layout-features/snap-layout-slide-transitions/).

Snap-layouts are designed for exactly the purpose you first imagined - they give you (the presentation author) the ability to position content completely independent of whatever standard content might be rendered on the slide.

For example, let's assume you want to add a markdown `#` (H1)  to the `north-east` corner of each slide in your presentations. You can not use plain markdown, so you would use the equivalent HTML snippet, as follows:

```
@snap[north-east]
<h1>The Heading</h1>
@snapend
```

That's it. You would then add this to every slide you want that heading to appear on. And it should not matter what content is being rendered on the individual slides, the heading should always appear in the exact same position on every slide.

And even more powerful way to take full control over the look of your custom heading would be to use a [custom css shortcut](https://gitpitch.com/docs/markdown-features/shortcuts) as follows:

```
@snap[north-east]
@css[slide-heading](The Heading)
@snapend
```

To use this you would need to define a custom *slide-heading* css class to style the font, spacing, colors, etc. of your heading.

As further reference, take a look at the GitPitch [Hello World](https://gitpitch.com/gitpitch/gitpitch) slideshow presentation demonstrates this exact behavior. 

Slide [#4](https://gitpitch.com/gitpitch/gitpitch#/4), slide [#6](https://gitpitch.com/gitpitch/gitpitch#/6), slide [#8](https://gitpitch.com/gitpitch/gitpitch#/8), slide [#11](https://gitpitch.com/gitpitch/gitpitch#/11) etc all use the same *north-east* snap-layout positioning for their heading and that positions stays perfectly constant in all cases, regardless of the specific content on individual slides. Which is exact what everyone should see if they are using snap-layouts.

Snap-layouts are a brand new feature and I'm wide open to feedback like yours on problems, bugs etc. So thanks again. I will be very interested to learn more about the problem cases you have described.

**Comment by joachim-n** [comment]
> At this point I'm not sure if you concerns are theoretical or if you have actually tried to use snap-layouts and you experiencing the problem you describe here - headings moving up and down? 

You've misunderstood me. I originally was not using snap layouts. I had this sort of thing:

```
# Common heading

- A list
- list 2

---

# Common heading

- Longer list that takes up more space
- more
- more
- more
- more
```

I wanted the effect that one list would be replaced with another, while 'Common heading' would stay fixed. However, the layout system places the 'Common heading' heading in a different place on the two slides, because the 2nd slide has a longer list.

Therefore, I looked to snap layouts as a solution: I wanted to place 'Common heading' in a snap north, so it's guaranteed to stay in the same place over multiple slides.

>  And it should not matter what content is being rendered on the individual slides, the heading should always appear in the exact same position on every slide.

Yup, that's working fine. It's just that having to use HTML there instead of Markdown syntax is a bit of a pain.

**Comment by gitpitch** [comment]
> I wanted the effect that one list would be replaced with another, while 'Common heading' would stay fixed. However, the layout system places the 'Common heading' heading in a different place on the two slides, because the 2nd slide has a longer list.

Just fyi, even without the use of snap-layouts, if you wanted all of your headings to appear in a fixed position on every slide you could have set the [automatic layout policy](https://gitpitch.com/docs/settings/layout/) for you presentation to use any of the  **top** layout options  - *top*, *top-left*, *top-right* - and it would not matter how long your list of items was the heading would always have been anchored in the same place.

Anyway, now you are spoiled for options. Enjoy.

**Comment by gitpitch** [comment]
UPDATE Nov 2018: GitPitch now supports [plain markdown](https://gitpitch.com/docs/layout-features/examples-plain-markdown) content inside of snap-tags. 

This update addresses the original issue raised on this thread. You can now use regular markdown syntax to create h1, h2, h3, etc headings or apply bold or italics etc  within your snap tags. Details in the [docs](https://gitpitch.com/docs/layout-features/examples-plain-markdown).

**Comment by RichiCoder1** [comment]
@gitpitch I'm guess fenced fragments (not `@code`) don't work?

**Comment by gitpitch** [comment]
Fenced code blocks are not supported with snap regions. As you noted, the `@code` markdown shortcut has been provided to solve that problem among other things. If for any reason you didn't want to use that shortcut you could try directly using `<pre><code>...</code></pre>` tags.
